src/core/messaging-service.js

```diff
@@ -207,7 +215,11 @@ export default class MessagingService {
       throw new Error(`Unknown destination ${destName}`)
     }
 
-    this._destinations.get(destName).registerReceiverCallback(callbackFn)
+    let destination = this._destinations.get(destName) // eslint-disable-line prefer-const
+    if (!destination.ableToReceive) {
+      throw new Error('Destination is not configured to receive messages')
+    }
+    destination.registerReceiverCallback(callbackFn)
```

Comment:
as far as I can see there no longer a registerReceiverCallback function in the Destination class as it has been renamed to registerReponseCallback

Reply:
Oh, it's a good catch! The function that calls `destination.registerReceiverCallback()` is no longer needed because the callback is registered in a constructor of a Destination subclass: https://github.com/alpheios-project/alpheios-messaging/blob/67734a656c5b850b27e6a41d9b544b6e96e88256/src/destinations/window-iframe-destination.js#L46-L53.
Thus, I removed this piece of code.

Reply:
In the PR that first added the messaging service to Treebank Template, the `registerReceiverCallback` function is being called ([here](https://github.com/perseids-publications/treebank-template/pull/15/files#diff-879b3b755ef22303729ed27c6d439cb4R70)). Replacing it with `registerResponseCallback` also does not work.

Using this new version of the code how would I write code equivalent to `this.service.registerResponseCallback(name, this.messageHandler);`?

Reply:
Sorry to make this breaking change. That's the reason why the version was changed to 2.0.0. The logic behind this change was that the use of `registerReceiverCallback()` was obligatory if someone wanted to listen to messages, but it was easy to omit its registration and then things will just not be working silently.

To fix this, I decided to move registration of a callback to the Destination's constructor. It is obligatory, if Destination has a RECEIVE mode enabled. Here is an example of how to register it: https://github.com/alpheios-project/lexis-cs/blob/master/src/cedict-service/service.js#L88-L91

What do you think? Does it make sense? If there is any better way to do so then we can make amends

Reply:
If I understand correctly, calls to `registerResponseCallback` should essentially be replaced with instantiating a new object? So the code I mentioned above would be changed to look like [this](https://github.com/perseids-publications/treebank-template/pull/19/files#diff-879b3b755ef22303729ed27c6d439cb4).

> What do you think? Does it make sense? If there is any better way to do so then we can make amends

I think either way makes sense. The opinion I lean towards is that doing it the way it is now, in the constructor, makes it a little easier for static webpages to include the code. On the other hand, the previous way of doing is slightly easier for frameworks like React or Vue which support code organization where the object is instantiated in one place but activated or deactivated it depending on the view being rendered. But either way would work.

The thing that I would find most useful for using the library in the `treebank-template` codebase isn't a particular architecture but rather a set of explicit code samples in the README. I hope that you consider adding something like that once the architecture is settled.

Reply:
that's a good suggestion. I will make an issue to make sure we don't forget to do it. Thank you!

Reply:
> If I understand correctly, calls to `registerResponseCallback` should essentially be replaced with instantiating a new object? So the code I mentioned above would be changed to look like [this](https://github.com/perseids-publications/treebank-template/pull/19/files#diff-879b3b755ef22303729ed27c6d439cb4).

Yes, this is how it should be used. 

> On the other hand, the previous way of doing is slightly easier for frameworks like React or Vue which support code organization where the object is instantiated in one place but activated or deactivated it depending on the view being rendered. But either way would work.

In situations like the one described above an alternative could be to create a messaging service with no destinations first and then add a destination using a `registerDestination()` method. It will activate that particular destination. Not exactly the same as before but probably somewhat close.

> The thing that I would find most useful for using the library in the treebank-template codebase isn't a particular architecture but rather a set of explicit code samples in the README. I hope that you consider adding something like that once the architecture is settled.

That's a great idea! Will do that a little later.